client/hotrod-client/src/main/java/org/infinispan/client/hotrod/StreamingRemoteCache.java

```diff
@@ -0,0 +1,67 @@
+package org.infinispan.client.hotrod;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * StreamingRemoteCache implements streaming versions of most {@link RemoteCache} methods
+ *
+ * @author Tristan Tarrant
+ * @since 9.0
+ */
+public interface StreamingRemoteCache<K> {
+   /**
+    * Retrieves the value of the specified key as an {@link InputStream}. It is up to the application to ensure
+    * that the stream is consumed and closed. The marshaller is ignored, i.e. all data will be read in its
+    * raw binary form. The returned input stream implements the {@link VersionedMetadata} interface.
+    *
+    * @param key      key to use
+    */
+   InputStream get(K key);
```

Comment:
If the `InputStream` implements the `VersionedMetadata` interface, shouldn't the signature be the following one instead?

    <T extends InputStream & VersionedMetadata> T get(K key);